Cookbook/Technical-Documents/BackendFeatureHowTo.md

```diff
@@ -0,0 +1,204 @@
+
+Essentially, it is defined like this:
+``` swift
+struct BackendResource<Request, Response, Auth> {
+    let path: ResourcePath         // URL
+    let method: Resource.Method    // GET, POST…
+    …
+    var headers: [String: String]
+}
+```
+
+As you can see, it contains everything that the network layer may need to make a network request: 
+* a request type that we provide if it's a `POST` endpoint
+* the type of the response we expect (that we just defined above)
+* the auth provider (more on that later)
+* the URL
+* request method
+* any additional headers we want to add
+
+A *service* that we are about to define is responsible for generating a `BackendResource` for each request that we want to make.
+
+### Service
+The purpose of a `Service` is to provide a `BackendResource` to execute network requests. As you may have guessed, this is also a simple struct, used as a namespace.
+
+As a rule, `GET` requests are named starting with `fetch`:
+``` swift
+struct CitiesService {
+    static func fetchCitiesResource() -> BackendResource<Void, CitiesResponse, ConcreteAuth> {
+        let citiesPath = "/v1/cities"
+        return BackendResource(
+            path: .api(.core, citiesPath),
+            method: .GET
+        )
+    }
+}
+```
+
+That's it!
+
+The `Service` usually provides `static` functions with parameters needed to create a request. In this case, we use `Void` in place of `Request` because it's just a `GET`, but we specify `CitiesResponse` as that's what we expect from the backend. 
+
+The URL is defined as a path, which is then constructed into a full URL with a helper enum. This accommodates the use of different environments while not requiring any changes from the clients.
+
+`ConcreteAuth` is the type of auth this endpoint expects as there are several, for different backends.
+
+Now that we have our `BackendResource`, we can define a *business controller* which will tie everything together and make the network call.
+
+However, first we need to learn about accessing backend resources.
+
+### Accessible
+`Accessible` is a protocol with just one function, which takes a backend resource and returns a signal producer with the parsed result:
+``` swift
+func access<Request, Response>(_ resource: BackendResource<Request, Response, NoAuth>, with request: Request) -> SignalProducer<Response, CoreError>
+```
+
+`CoreError` is preferred as it allows both network and parsing errors and is used throughout the network layer.
+
+You may have noticed `NoAuth` in the signature – this is only good for unauthenticated requests, like fetching data before the user logs in. After that, you'll likely want to use `AuthenticatedAccessible` which provides a slightly different method:
+
+``` swift
+func access<Request, Response>(_ resource: BackendResource<Request, Response, ConcreteAuth>, with request: Request) -> SignalProducer<Response, CoreError>
```

Comment:
Maybe it's worth introducing the two available concrete auth  types: KongAuh and ClincialAuth.
Also an important thing is how you construct your resource using the KongAuth.patientId.

Reply:
I thought this was a little out of scope (also exposing unnecessary internal details), and anyway auth methods are probably going to change going forward with the work that @jtomanik is doing.

Reply:
At minimum I suggest in removing the ConcreteAuth since there is no such type. 
Otherwise this will confuse even more and the reader would need at least another document to build a feature that works with backend.

Reply:
Ah, confusing the reader, now that makes sense. I'll see what I can do about it. I just didn't want to  complicate the doc too much.